### DISCUSSION TITLE
Getting metadata of url-stream but *Not* with player

### DISCUSSION THREAD

**Discussion by DrNeuroSurg** [pschatzmann/arduino-audio-tools#1041] [Q&A] [answered]
I'm working on a web-radio (listener) ... (almost) everything is working - thanks for this *GREAT* library.
I like to use your **MetaDataOutput** - class. But I'm not using the player :-(
I can play streams (mp3) - it works. 
I dont't want tu use the player-class because of volume ..

This is  part of my code:

..... **defines**
```

    MetaDataOutput outMeta; 
   URLStream urlStream(wifi, password);
   I2SStream i2s; // final output of decoded stream
   VolumeStream volume(i2s);

    EncodedAudioStream dec(&volume, new MP3DecoderHelix()); // Decoding stream

   StreamCopy copier(dec, urlStream); // copy url to decoder

.... **setup**

      i2s.begin(config);
      dec.begin();
      
      outMeta.setCallback(printMetaData);
      outMeta.begin();

      volume.begin(config);
      volume.setVolume(0.1);
      urlStream.begin("http://s6-webradio.antenne.de/antenne", "audio/mp3");

... **loop**

      copier.copy();
```

**Comment by pschatzmann** [comment, marked as answer]
You are not providing any data to the MetaDataOutput. You can do this e.g. with a MultiOutput like in [this example](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-stream/streams-sdfat_mp3-metadata/streams-sdfat_mp3-metadata.ino)

The player class has the volume control and metadata support already built in. There I am using a callback method of the StreamCopy to provide the data.

**Reply by DrNeuroSurg** [reply]
Thanks a lot - I'# give it a try  :-)
